### DISCUSSION TITLE
DEVSIM instalation and integration

### DISCUSSION THREAD

**Discussion by adophobr** [gdsfactory/gdsfactory#799] [Q&A] [answered]
Hi,

I was very happy when I learned that DEVSIM was integrated into gdsfactory (great work!). I updated it with the command:

`pip install gdsfactory --upgrade`

and then:

`pip install gdsfactory[full]`

I noticed that the DEVSIM package was installed:
![image](https://user-images.githubusercontent.com/15238847/199000692-793266b0-601a-4ad5-83bc-288fcfd03ce2.png)
However, the integration with gdsfactory doesn't works:
![image](https://user-images.githubusercontent.com/15238847/199000881-865e2a1a-55c8-40a2-9108-6e07286a74fd.png)
Also I tried to find some functions into gdsfactory.simulation related with DEVSIM:
![image](https://user-images.githubusercontent.com/15238847/199001853-4a97ef73-e56d-42ed-9a8f-6c8cea6d2366.png)
and I didn't find any function. Then I deleted and installed gdsfactory (and plugins) and the issues persisted.

Is there something I did wrong? Is there any step in the update that I didn't do?

**Comment by simbilod** [comment, marked as answer]
Hi @adophobr , happy to hear you like the project :)

I have found the issue and pushed a fix to #803 ; once this goes through and we have a new release you will be able to install as `pip install gdsfactory[devsim]`

Stay tuned for more updates; right now we have a hardcoded PIN waveguide simulation with coarse mesh, soon I will push a generic mesher for gdsfactory Component objects which should be much more flexible
